mlrun/datastore/sources.py

```diff
@@ -73,32 +73,38 @@ def get_table_object(self):
         """get storey Table object"""
         return None
 
-    def to_dataframe(self):
-        return mlrun.store_manager.object(url=self.path).as_df()
-
-    def filter_df_start_end_time(self, df, time_field):
-        # give priority to source time_field over the feature set's timestamp_key
-        if self.time_field:
-            time_field = self.time_field
+    def to_dataframe(self, **kwargs):
+        kwargs = self._ingest_init_time_params_to_kwargs(kwargs)
+        return mlrun.store_manager.object(url=self.path).as_df(**kwargs)
 
-        if self.start_time or self.end_time:
-            self.start_time = (
-                datetime.min if self.start_time is None else self.start_time
-            )
-            self.end_time = datetime.max if self.end_time is None else self.end_time
-            df = df.filter(
-                (df[time_field] > self.start_time) & (df[time_field] <= self.end_time)
-            )
-        return df
+    def _ingest_init_time_params_to_kwargs(self, args_dict):
```

Comment:
The word ingest already has an important meaning in the code base, and it's not really applicable here anyway.

More generally, I think this method is unnecessary. It's better to just pass these arguments by name where applicable, instead of messing around with the kwargs. E.g. `time_field=time_field or self.time_field`, a pattern that is very common in the code.